CRM/Price/BAO/LineItem.php

```diff
@@ -843,8 +845,9 @@ protected function getAdjustedFinancialItemsToRecord($entityID, $entityTable, $c
         if ($amountChangeOnTextLineItem !== (float) 0) {
           // calculate the amount difference, considered as financial item amount
           $updateFinancialItemInfoValues['amount'] = $amountChangeOnTextLineItem;
-          if ($previousLineItems[$updateFinancialItemInfoValues['entity_id']]['tax_amount']) {
-            $updateFinancialItemInfoValues['tax']['amount'] = $lineItemsToUpdate[$updateFinancialItemInfoValues['entity_id']]['tax_amount'] - $previousLineItems[$updateFinancialItemInfoValues['entity_id']]['tax_amount'];
+          if ($previousLineItem['tax_amount']
+            && $previousLineItems[$updateFinancialItemInfoValues['entity_id']]['tax_amount'] !== 0.00) {
```

Comment:
@eileenmcnaughton Do these 0.00 comparisons work? As a float the value would be 0.0, as a string it could be "0.00"

Reply:
I added them because the tests failed without them - it's casting to float now